tests/unit/test_charm.py

```diff
@@ -26,3 +126,22 @@ def test_update_status_success(self, *_):
 
         self.harness.charm.on.update_status.emit()
         self.assertEqual(self.harness.charm.unit.status, ActiveStatus("slurmrestd available"))
+
+    @patch("pathlib.Path.read_text")
+    @patch("ops.model.Unit.set_workload_version")
+    def test_upgrade_fail(self, *_):
+        self.harness.charm.on.upgrade_charm.emit()
+        self.assertEqual(self.harness.charm.unit.status, BlockedStatus("Error installing slurmrestd"))
+
+    @patch("pathlib.Path.read_text")
+    @patch("ops.model.Unit.set_workload_version")
+    @patch("slurm_ops_manager.SlurmManager.needs_reboot", new_callable=PropertyMock(return_value=False))
+    @patch("interface_slurmrestd.SlurmrestdRequires.is_joined", new_callable=PropertyMock(return_value=True))
+    @patch("interface_slurmrestd.SlurmrestdRequires.get_stored_munge_key", lambda _: True)
+    @patch("interface_slurmrestd.SlurmrestdRequires.get_stored_jwt_rsa", lambda _: True)
+    @patch("interface_slurmrestd.SlurmrestdRequires.get_stored_slurm_config", lambda _: True)
+    def test_upgrade_success(self, *_):
+        self.harness.charm._stored.slurm_installed = True
+
+        self.harness.charm.on.upgrade_charm.emit()
+        self.assertEqual(self.harness.charm.unit.status, ActiveStatus("slurmrestd available"))
```

Comment:
Same with testing install success, do we want to be patching out the fact that the version file is missing? Might lull another contributor into a false sense of security that their copy of the charm is working as expected. I think you should not patch the version file and let the test fail. I believe that one thing we should be focused on while making our improvements to the charms is figuring out how we can get rid of this version file.

Reply:
I agree, no longer patching out version file.